src/hipscat_import/catalog/run_import.py

```diff
@@ -12,81 +12,67 @@
 from tqdm import tqdm
 
 import hipscat_import.catalog.map_reduce as mr
-import hipscat_import.catalog.resume_files as resume
 from hipscat_import.catalog.arguments import ImportArguments
 
 
 def _map_pixels(args, client):
     """Generate a raw histogram of object counts in each healpix pixel"""
 
-    raw_histogram = resume.read_histogram(args.tmp_path, args.mapping_healpix_order)
-    if resume.is_mapping_done(args.tmp_path):
-        return raw_histogram
-
-    mapped_paths = set(resume.read_mapping_keys(args.tmp_path))
+    if args.resume_plan.is_mapping_done():
+        return
 
+    reader_future = client.scatter(args.file_reader)
     futures = []
-    for file_path in args.input_paths:
-        map_key = f"map_{file_path}"
-        if map_key in mapped_paths:
-            continue
+    for key, file_path in args.resume_plan.map_files:
         futures.append(
             client.submit(
                 mr.map_to_pixels,
-                key=map_key,
+                key=key,
                 input_file=file_path,
-                file_reader=args.file_reader,
-                filter_function=args.filter_function,
+                cache_path=args.tmp_path,
+                file_reader=reader_future,
+                mapping_key=key,
                 highest_order=args.mapping_healpix_order,
                 ra_column=args.ra_column,
                 dec_column=args.dec_column,
             )
         )
 
     some_error = False
-    for future, result in tqdm(
-        as_completed(futures, with_results=True),
+    for future in tqdm(
+        as_completed(futures),
         desc="Mapping  ",
         total=len(futures),
         disable=(not args.progress_bar),
     ):
         if future.status == "error":  # pragma: no cover
             some_error = True
-            continue
-        raw_histogram = np.add(raw_histogram, result)
-        resume.write_mapping_start_key(args.tmp_path, future.key)
```

Comment:
Where does the start_key get written now?

Reply:
There's no more distinction between start/done. The two keys were needed in case there was a failure within the histogram writing. In that case, we would have marked the shard as complete, but its raw histogram would not contribute to the overall histogram.

The shard's raw histogram is now written within the map method, so we don't need to worry about that edge case anymore.

Reply:
Oh cool, thanks!